MSAL/src/instance/MSALDeviceInfoProvider.m

```diff
@@ -32,59 +32,91 @@
 #import "MSIDRequestParameters+Broker.h"
 #import "MSALDeviceInformation+Internal.h"
 
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+#import "MSIDRegistrationInformation.h"
+
 @implementation MSALDeviceInfoProvider
 
 - (void)deviceInfoWithRequestParameters:(MSIDRequestParameters *)requestParameters
-                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock API_AVAILABLE(ios(13.0), macos(10.15))
+                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock
 {
-    if (![requestParameters shouldUseBroker])
-    {
-        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorBrokerNotAvailable, @"Broker is not enabled for this operation. Please make sure you have enabled broker support for your application", nil, nil, nil, nil, nil, YES);
-        completionBlock(nil, error);
-        return;
-    }
-    
-    
-    if (![MSIDSSOExtensionGetDeviceInfoRequest canPerformRequest])
-    {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, requestParameters, @"Broker is not present on this device. Defaulting to personal mode");
-        
-        MSALDeviceInformation *msalDeviceInfo = [MSALDeviceInformation new];
-        msalDeviceInfo.deviceMode = MSALDeviceModeDefault;
-        completionBlock(msalDeviceInfo, nil);
-        return;
-    }
-    
-    NSError *requestError;
-    MSIDSSOExtensionGetDeviceInfoRequest *ssoExtensionRequest = [[MSIDSSOExtensionGetDeviceInfoRequest alloc] initWithRequestParameters:requestParameters
-                                                                                                                                  error:&requestError];
-    
-    if (!ssoExtensionRequest)
-    {
-        completionBlock(nil, requestError);
-        return;
-    }
-    
-    if (![self setCurrentSSOExtensionRequest:ssoExtensionRequest])
+    MSALDeviceInformation *deviceInformation = [MSALDeviceInformation new];
```

Comment:
shouldUseBroker should still be the first check. If it's not enabled, we shouldn't proceed.

Reply:
It is observed "[requestParameters shouldUseBroker]" returns false if Authenticator is not installed on the device.
IMO we should return device registration info, irrespective of who registered the device(CP, Authenticator etc).